addons/website_slides/models/slide_slide.py

```diff
@@ -1285,6 +1285,32 @@ def _default_website_meta(self):
     # Data / Misc
     # ---------------------------------------------------------
 
+    def is_category_completed(self):
+        """ Determine if this slide category is completed.
+        For the category to be completed, each one of its slides need to have a slide.slide.partner record and be completed.
+        """
+        if not self.category_id:
+            return False
+        current_category_slides_info = self.env['slide.slide.partner'].sudo().search(
+            [('slide_id', 'in', self.category_id.slide_ids.ids), ('partner_id', '=', self.env.user.partner_id.id)]
+        )
+        partial_category_completion = len(current_category_slides_info) != len(self.category_id.slide_ids.ids)
+        return not partial_category_completion and all([slide.completed for slide in current_category_slides_info])
+
+    def get_next_category_id(self):
+        channel_sorted_categories = list(self.channel_id.slide_ids.filtered(lambda s: s.is_category).sorted(key=lambda s: s.sequence).ids)
```

Comment:
why do you need to sort manually ?

With the order on the model it should be enough no ? (if you want the inverse order, just inverse the logic here)

```
_order = 'sequence asc, is_category asc, id asc'
```